pinot-broker/src/main/java/org/apache/pinot/broker/routing/RoutingManager.java

```diff
@@ -354,9 +359,12 @@ public synchronized void buildRouting(String tableNameWithType) {
       }
     }
 
+    QueryConfig queryConfig = tableConfig.getQueryConfig();
+    Long timeoutMs = queryConfig != null ? queryConfig.getTimeoutMs() : null;
```

Comment:
Also, I notice in the test you specified `TimeoutMs` to be 0. Does 0 has special meaning? If yes, could you add it to the description of this variable?

Reply:
It should not be `0` (added a check), but `1ms` might be too long for broker compilation to timeout. Let me revise the test, merge the broker & server timeout so that test is not flaky.